### PR TITLE
[]MLN-739] Implemented news-letter checkbox in sign-up page

### PR DESCRIPTION
**Pull request by gonuguntla**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -104,4 +104,54 @@ function validateNewsLetterEmail(event) {
...
+
+function veriryNewsLetterEmailInSignUpPage(val, email, alt_email) {
+  // If alternate email is present in sign-up page take alternate-email to send news-letters other-wise DOE email.
+  news_letter_email = alt_email
+  if (news_letter_email == ""){
+    news_letter_email = email
+  }
+  $('#news-letter-ajax-message').show();
+  $('#news-letter-ajax-message').html('Verifying email')
+  $('#news-letter-ajax-message').css("color", '#c4262d');
```
Comment (daryanypl) [inline app/assets/javascripts/application.js:117]: This looks like a dark red for the message letters, which is in our color scheme, but might not be super-visible.  Maybe a good place to take note for the future redesign, I'd love to see our error messaging be easier to absorb.

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -104,4 +104,54 @@ function validateNewsLetterEmail(event) {
...
+  $("#sign_up_button").prop("disabled", false);
+  $("#sign_up_button").css("background-color", "grey");
+
+  // Ajax call to validate the news-letter email.
+  $.ajax({ type: "POST", url: "/news_letter/validate_news_letter_email_from_user_sign_up_page", contentType: "application/json; charset=utf-8",
+           data: JSON.stringify({email: news_letter_email, sign_up_page: true}), dataType: "json", success: function (result) {
+
+    if ($('input[name=news_letter_email]').is(':checked')) {
+      if (result.success == "success") {
```
Comment (daryanypl) [inline app/assets/javascripts/application.js:126]: What kinds of email validation are we doing?  Anything security-related?

**Review by daryanypl — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -104,4 +104,54 @@ function validateNewsLetterEmail(event) {
     $('#nl_message').html("");
     $('#nl_message').hide();
    }
+}
+
+function veriryNewsLetterEmailInSignUpPage(val, email, alt_email) {
```
[the remark this answers is not in the document]
Comment (gonuguntla) [reply app/assets/javascripts/application.js]: @daryanypl  Thanks. fixed typo.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (gonuguntla) [reply app/assets/javascripts/application.js:117]: Yes will take a note

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (gonuguntla) [reply app/assets/javascripts/application.js:126]: this is already existing new-letter validation only. Just checking the format of email and checking email is there in google sheets or not.

**Review by daryanypl — APPROVED** [review]

**Pull request merged**